### PR TITLE
feat: Add documentation for the new directory structure

### LINKED ISSUE #91: Refer to "blocks" as "procedures"
Matches the language of jsPsych and Honeycomb in v3.3

### LINKED ISSUE #92: Update folder hierarchy docs

### PR DESCRIPTION
**Pull request by RobertGemmaJr**
- Adds documentation for the new `src/experiment/` folder
  - `index.js` and `honeycomb.js`
  - `experiment/procedures/`
  - `experiment/trials/`
- Better explains how the `honeycomb_`.js files are for the example experiment and that the user should create their experiment in new files
- Adjusts the documentation for the new organization of electron files
- Better "tabs" the callouts so they're in line with the piece of documentation their talking about

### CODE REVIEW

**Comment by RobertGemmaJr** [comment]
This is going to be a massive comment 😬
image
image
image
image
image
image

File: docs/project_organization/directory_structure.mdx [hunk 1]
```diff
@@ -38,149 +38,184 @@ This folder contains different files used to pass environment variables (setting
 ## node_modules/
 
 :::danger
-`node_modules/` is written to when running `npm install` and should never be manually edited.
+`node_modules/` is written to when running `npm install` and should never be edited. It is in Honeycomb's `.gitignore` and should never be added to git.
 :::
 
 ## psiturkit/
 
 The file `psiturk-it` inside `psiturkit/` is a bash script used to instal PsiTurk locally - see [PsiTurk](psiturk#build-instructions) for more information.
```
Comment (tdivoll) [inline docs/project_organization/directory_structure.mdx:46]: Should you add the extension to this `psiturk-it` file to better distinguish between the file and the directory?

File: docs/project_organization/directory_structure.mdx [hunk 2]
```diff
@@ -38,149 +38,184 @@ This folder contains different files used to pass environment variables (setting
...
-- `trigger.js` for equipment-related settings (e.g. event markers). It uses a slightly different style of Javascript as it is imported both in the React app as well as the electron process.
-- `config.json` contains the settings for your task. Usage of the config file allows for easy updating of task settings. Common settings can be written once in the config file and re-used throughout the task.
-- `language.json` contains the language used in your task. This files allows for easy internationalization of the task (e.g. english and spanish) and mturk-specific language. Common phrases can be written once in a language file and re-used throughout the task.
-
-### lib/
-
-A library of utility and markup are located here. This allows for functions and html to be re-used wherever needed.
-
-#### markup/
+- `language.json` contains the language used in your task. This file allows for easy internationalization of the task (e.g. english and spanish) and mturk-specific language. Common phrases can be written once and re-used throughout the task.
```
Comment (tdivoll) [inline docs/project_organization/directory_structure.mdx]:
```suggestion
- `language.json` contains the language used in your task. This file allows for easy internationalization of the task (e.g. English and Spanish) and mturk-specific language. Common phrases can be written once and re-used throughout the task.

```

File: docs/project_organization/directory_structure.mdx [hunk 3]
```diff
@@ -38,149 +38,184 @@ This folder contains different files used to pass environment variables (setting
...
+- `main.js` contains the global settings (e.g. whether Honeycomb is running online or in the clinic) passed from env variables and logic for loading the appropriate files.
```
Comment (tdivoll) [inline docs/project_organization/directory_structure.mdx]:
```suggestion
- `main.js` contains the global settings (e.g., whether Honeycomb is running online or in the clinic) passed from env variables and logic for loading the appropriate files.

```

File: docs/project_organization/directory_structure.mdx [hunk 4]
```diff
@@ -38,149 +38,184 @@ This folder contains different files used to pass environment variables (setting
...
 - `fullscreen.js` contains trials for entering and exiting fullscreen mode.
 - `holdUpMarker.js` prompts the user to connect their event marker and hold it up to the camera.
-- `quizTrial.js` contains trials for a quiz/survey.
+- `honeycombTrials.js` contains the individual trials used in the Honeycomb task. These trials are imported into `experiment/procedures/honeycombProcedure.js` and `experiment/honeycomb.js`.
+  :::tip
+  These trials are for an example experiment! Be sure to write trials pertaining to your task in their own file(s).
+  :::
+- `introduction.js` displays a message to the user welcoming them to the experiment. They must click on a button to continue.
+- `name.js` displays the name of the task to the user.
 - `startCode.js` emits a start code to a photodiode spot and audible beep
```
Comment (tdivoll) [inline docs/project_organization/directory_structure.mdx]:
```suggestion
- `startCode.js` emits a start code to a photodiode spot and audible beep.

```

File: docs/project_organization/directory_structure.mdx [hunk 5]
```diff
@@ -38,149 +38,184 @@ This folder contains different files used to pass environment variables (setting
...
+- `tags.js` contains functions for wrapping [language](#config-1) in common html tags. You should always wrap your language in a tag to ensure it is displayed correctly.
+  - For example: `p('Hello World')` will return `<p>Hello World</p>`.
+
+:::tip
+The `tag` function inside `tags.js` can be used to wrap language in any html tag you need.
+:::
 
 ## Other Folders/Files
 
 - `.nvmrc` determines which version of node that Honeycomb is designed to be run on
```
Comment (tdivoll) [inline docs/project_organization/directory_structure.mdx]:
```suggestion
- `.nvmrc` determines which version of node that Honeycomb is designed to be run on.

```

File: docs/project_organization/directory_structure.mdx [hunk 6]
```diff
@@ -38,149 +38,184 @@ This folder contains different files used to pass environment variables (setting
...
-- `.github/workflows/` contains .yaml files used to build Honeycomb on a pull request (`build.yaml`) and create task executables either in a single instance (`package.yaml`) or for a full release (`release.yaml`)
+- `.github/workflows/` contains .yaml files used for [CI/CD](ci_cd) with GitHub Actions
```
Comment (tdivoll) [inline docs/project_organization/directory_structure.mdx]:
```suggestion
- `.github/workflows/` contains .yaml files used for [CI/CD](ci_cd) with GitHub Actions.

```

File: docs/project_organization/directory_structure.mdx [hunk 7]
```diff
@@ -38,149 +38,184 @@ This folder contains different files used to pass environment variables (setting
...
 - `package.json` contains metadata about your project, a list of the dependencies needed for the project, and scripts to run tasks related to your task. The [Quick Start](quick_start#4-change-name-and-description) lists which metadata should be changed.
 - `cli.mjs` is the script used to download and delete data stored in Firestore.
 - `version.js` is the script used to keep track of which version of the task a given experiment is using
```
Comment (tdivoll) [inline docs/project_organization/directory_structure.mdx]:
```suggestion
- `version.js` is the script used to keep track of which version of the task a given experiment is using.

```

**Review by tdivoll — APPROVED** [review]
looks good! Just a few little cleanups for consistency.

File: docs/project_organization/directory_structure.mdx (hunk 1 above)
Comment (RobertGemmaJr) [reply docs/project_organization/directory_structure.mdx:46]: Eh, I'm not to worried about it. Nobody really uses psiturk anymore and it's going to be deprecated in v4

**Pull request merged**